src/contracts/UpgradeRegressionManager.sol

```diff
@@ -416,4 +420,51 @@ contract UpgradeRegressionManager is IUpgradeRegressionManager {
     emit AdminSet(admin, _newAdmin);
     admin = _newAdmin;
   }
+
+  /// @notice Returns the current state of a proposed rollback.
+  /// @param _rollback The rollback to check.
+  /// @return The current ProposalState of the rollback, which can be:
+  /// - `Unknown`: rollback does not exist (i.e., never proposed).
+  /// - `Pending`: proposed but not yet queued and not expired.
+  /// - `Expired`: proposed but not queued before expiration.
+  /// - `Queued`: queued for execution, but delay not elapsed.
+  /// - `Active`: queued and ready for execution.
+  /// - `Executed`: rollback has already been executed (terminal state).
+  /// - `Canceled`: rollback was canceled before execution (terminal state).
+  /// @dev This function determines the rollback's lifecycle state based on timestamps and flags.
+  function _getState(Rollback memory _rollback) internal view returns (ProposalState) {
+    // Revert if the rollback was not proposed (i.e., does not exist).
+    if (_rollback.queueExpiresAt == 0) {
+      return ProposalState.Unknown;
+    }
+
+    // Check if the rollback has been executed.
+    if (_rollback.executed) {
+      return ProposalState.Executed;
+    }
+
+    // Check if the rollback has been canceled.
+    if (_rollback.canceled) {
+      return ProposalState.Canceled;
+    }
+
+    // Check if the rollback has been queued for execution.
+    if (_rollback.executableAt != 0) {
+      if (block.timestamp >= _rollback.executableAt) {
+        // Rollback's execution time has arrived.
+        return ProposalState.Active;
```

Comment:
I'm not sure how I feel about this semantically. In the Governor:

* "Active" means the proposal is being voted on
* A proposal that is ready for execution is still just in the `Queued` state

We don't have voting, so my expectation would be that we would not have an "Active" state, and that a proposal that is ready for execution, but not yet executed, would still be queued.

Is there a strong reason we shouldn't stick with the Governor convention, here? I can see how it might be nice to have a separate state for "ready to be executed" but, as implemented this seems confusing.

Reply:
I think it's arguable whether "Ready" is semantically a separate state. One way to look at it is: the proposal *is* still Queued, it's just that it is now possible for the state to *change*.

Comparing the `executableAt` entry in the Rollback structure to the current timestamp allows the integrator to figure out if it is in fact executable. If we wanted, we could add a public view helper method like `isExecutable` or something like that to help out with this. I'd be good with pulling that into it's own issue.

Reply:
FYI created an issue for this here: https://github.com/ScopeLift/governance-rollback/issues/54

Otherwise this is good to resolve.

Reply:
Ah thanks for that :D